### PR TITLE
Rename constructor `Proof_level.{None -> No_check}`

### PR DESCRIPTION
**Pull request by georgeee**
Using widely used `None` for name of constructor lead to an issue when `~proof_level` parameter changed its type and compiler left it unnoticed.

Explain how you tested your changes:
* [x] Nightly passes

Checklist:

- [x] Dependency versions are unchanged
  - Notify Velocity team if dependencies must change in CI
- [ ] Modified the current draft of release notes with details on what is completed or incomplete within this project
- [x] Document code purpose, how to use it
  - Mention expected invariants, implicit constraints
- [x] Tests were added for the new behavior
  - Document test purpose, significance of failures
  - Test names should reflect their purpose
- [x] All tests pass (CI will check this if you didn't)
- [x] Serialized types are in stable-versioned modules
- [x] Does this close issues? None

### CODE REVIEW

**Comment by georgeee** [comment]
!ci-nightly-me

**Comment by georgeee** [comment]
!ci-build-me

**Comment by svv232** [comment]
I have attempted to fix: https://github.com/MinaProtocol/mina/pull/16106, and will re-run the nightly on that branch.

**Comment by svv232** [comment]
that branch passes the nightly https://buildkite.com/o-1-labs-2/mina-end-to-end-nightlies/builds/2563#_

I do think it took some massaging. The following observations were made:

- libp2p unit tests occasionally feels because of dropped peers; i am not sure whether this is caused by connection mismanagement, but considering this failure isn't deterministic I would guess something along these lines. The goroutine logic in the unit tests can also be hard to follow at times.

- integration tests occasionally fail with docker swarm issues i.e lack of connections; i'm not sure if this is caused by network load or buildkite agent state, but it definitely makes retrying necessary often.

- fuzzy zkapp unit tests
I am actually not sure about this one, but i have noticed a few failures in the logging with passing tests regardless. This definately requires further investigation.

**Comment by georgeee** [comment]
Branch includes a lightnet fix to make nightly pass

**Comment by georgeee** [comment]
!ci-build-me

**Comment by georgeee** [comment]
!ci-nightly-me

**Review by svv232 — APPROVED** [review]

File: scripts/docker/helper.sh [hunk 1]
```diff
@@ -43,6 +43,18 @@ function export_suffixes () {
             ;;
         esac
         ;;
+        lightnet)
```
Comment (mrmr1993) [inline scripts/docker/helper.sh:46]: This seems orthogonal to the PR; can you split it out?

Comment (georgeee) [reply scripts/docker/helper.sh:46]: Agree, Dariusz just created a PR for that 🎉 

https://github.com/MinaProtocol/mina/pull/16084

**Comment by mrmr1993** [comment]
!ci-build-me

**Review by mrmr1993 — APPROVED** [review]

**Review by SanabriaRusso — APPROVED** [review]

**Comment by dannywillems** [comment]
I suppose it can be merged as everyone approves. Doing so.
